src/scripts/main.js

```diff
@@ -1,11 +1,23 @@
 'use strict';
 
 function waitFor(element, eventName) {
-// write your code here
-}
+  const result = new Promise((resolve, reject) => {
+    document.addEventListener(eventName, (e) => {
+      if (e.target === element) {
+        resolve(`It was ${eventName} on the element:
+        ${element.nodeName}, id: ${element.id}.`);
+      }
+    });
+  });
+
+  return result;
+};
 
 const printMessage = (message) => {
-// write your code here
+  const body = document.getElementById('document');
```

Comment:
The body of the document is usually selected by `const body = document.querySelector('body');` instead of `document.getElementById('document')`